### ISSUE TITLE
Angular Guide Fails to Build in Prod Mode

### ISSUE THREAD

**Issue by craigsumner** [cerebral/overmind#351]
I'm exploring this library for use with angular, but I'm encountering a build issue I'm struggling to understand. I can only reproduce the issue with angular's 'prod' build mode, and not in normal, or 'dev' build mode.

I've reproduced the problem with multiple versions of angular and overmind, in a small variety of local projects. The following repo, which was created by another user for a different issue, can also reproduce the problem:

https://github.com/ReneCode/ang-overmind

The overmind usage seems consistent with what's illustrated in the guide here:

https://overmindjs.org/guides/beginner/09_usingovermindwithangular?view=angular&typescript=true

Just change line 7 of package.json from this:

`    "build": "ng build",`

to this:

`    "build": "ng build --prod",`

Then run:

`npm run build`

The error is:

`
ERROR in overmind-angular/lib/directive.ts(3,22): Error during template compile of 'OVERMIND_INSTANCE'
  Only initialized variables and constants can be referenced in decorators because the value of this variable is needed by the template compiler.
overmind-angular/lib/directive.ts(3,22): Error during template compile of 'OVERMIND_INSTANCE'
  Only initialized variables and constants can be referenced in decorators because the value of this variable is needed by the template compiler.
Invalid provider for the NgModule 'AppModule in /ang-overmind/src/app/app.module.ts' - only instances of Provider and Type are allowed, got: [?[object Object]?, ...]
`

The documentation for that error is here:

https://angular.io/guide/aot-metadata-errors#only-initialized-variables

I've tried to resolve this by using a factory provider instead of a value provider for `OVERMIND_INSTANCE`, but I'm still struggling.

https://angular.io/guide/dependency-injection-providers#factory-providers

I'd like to see any example overmind-angular project which compiles in production mode.
Can anyone point me to one?

**Comment by craigsumner** [comment]
I wanted to understand this better, so I tried pulling the 'overmind-angular' source directly into the 'ang-overmind' project, and using it directly, instead of via the npm artifact.

I copied the three files located here:

https://github.com/cerebral/overmind/tree/next/packages/node_modules/overmind-angular/src

And changed the relevant imports in the 'app.module.ts' and 'overmind/index.ts' files.

My intention was to try and isolate the problem by fiddling with the code, but I discovered something I didn't expect before I got that far. The above steps alone resolved the build error.

This seems like a potentially useful clue, but it's still not clear to me what the build error specifically wants, or how using the original typescript source provides it.
